Modules/socketmodule.h

```diff
@@ -76,6 +76,12 @@ struct SOCKADDR_BTH_REDEF {
 # else
 typedef int socklen_t;
 # endif /* IPPROTO_IPV6 */
+
+/* Future remove once Py_WINVER has been bumped to >=0x0604 */
```

Comment:
Nit: Up to Steve to decide, but I would prepare a preprocessor warning to trigger when Py_WINVER has been raised above 0x0604, so we don't forget about this 😎

Reply:
Any example on how to create the preprocessor warning? I'm assuming it would be essentially

```c
# ifdef AF_HYPERV
Somehow create the warning here
# else
#  define AF_HYPERV 34
# endif
```

Reply:
I believe `#warn Message` is what you want (it might be `#warning`)

Reply:
I never remember which it is; I seldom use it. IIRC, it is `#warning` (and also `#error` if you want to halt compilation). But this is a case where one might _consider_ using it. I'm not saying we should, though :)

Reply:
So you could do something like this:

```c
#if Py_WINVER >= 0x0604
#  warning "Remove some of the stuff below this line"  // <= write something more sensible here
#else  // put the stuff we need now, but might not need in the future here:
#  ifndef AF_HYPERV
#    define AF_HYPERV 34
#  endif
#endif
```

Reply:
Looks like warning isn't valid for MSVC and I can't really find an equivalent. I could do this though

```c
#if Py_WINVER >= 0x0604
#pragma message ( "AF_HYPERV should be defined, remove the ifdef dance in socketmodule.h" )
#else
#  ifndef AF_HYPERV
#    define AF_HYPERV 34
#  endif
#endif /* Py_WINVER */
```

It will print the message in the build output but it just won't fail the build if warnings are treated as errors.

Reply:
Up to Steve if we want to put something like this in there.

Reply:
I think it can probably just be a comment. Include `Py_WINVER >= 0x0604` in the comment though, so whoever goes searching for things to clean up will find it.

Reply:
A comment has been added that include `Py_WINVER >= 0x0604` to the header. Thanks for the info.